src/Core/Framework/Api/ApiDefinition/Generator/Schema/AdminApi/paths/consent.json

```
@@ -0,0 +1,109 @@
+{
+  "openapi": "3.1.0",
+  "info": [],
+  "paths": {
+    "\/consents": {
+      "get": {
+        "tags": [
+          "Consent Management"
+        ],
+        "summary": "List all consents for current user",
+        "description": "Returns a list of all available consents with their current status for the authenticated admin user. Each consent includes its name, identifier, and current state (requested, accepted, or revoked).",
+        "operationId": "fetchConsents",
+        "responses": {
+          "200": {
+            "description": "List of consents with their status",
+            "content": {
+              "application\/json": {
+                "schema": {
+                  "$ref": "#\/components\/schemas\/consentListResponse"
+                },
```

This is for comment above with keeping it in the endpoint as we're not reusing it
```suggestion
                "schema": {
                  "type": "array",
                  "description": "List of consent states for the authenticated user",
                  "items": {
                    "$ref": "#/components/schemas/ConsentState"
                  }
                },
```

---

Additionally - how many consents there can be? Would pagination make sense here?

---

Imo when we need pagination we have too many consents. But say there is 20 users for 5 consents - there could be 100 states. I want to keep it simple at first really, it's all internal, but of course we don't know how usage will be in the wild and for a system with many admin users there could be a lot of data. If that's true we need to refactor quite a bit, because we also keep all states in memory atm for simplicity. Do you have some thoughts @keulinho ? Is it okay to keep it naive and if we get some reports or notice performance issues ourselves, improve it at a later stage?

---

Then it's okay not to optimise too quickly, the proposition was to have like 
```json
{
  consents: []
}
```

and then you'd add
```json
{
  consents: [],
  limit: 15,
  offset: 30,
  count: 150
}
```
then migration might be a bit easier (for example instead of breaking consent flow you just see not all of them by default)

With array it's full BC, not progressive transition.
But that was definetly additional question, nothing blocking :)

---

Makes total sense but we also wanna make the whole thing internal for now because we expect we will iterate on it :)

---

och yeah, then definetly `Experimental` tag is needed on these! Otherwise it's not that easy to change later ;)